### PR TITLE
Implement Helicopter subclass

### PR DESCRIPTION
**Pull request by tahg**
This should finish the Helicopter source file. I've tried all the usual suspects and can't get the ordering correct, so no idea what's going in this function.

### CODE REVIEW

File: LEGO1/helicopter.cpp [hunk 1]
```diff
@@ -321,8 +321,51 @@ void Helicopter::VTable0x70(float p_float)
...
+// FUNCTION: LEGO1 0x100040a0
 MxResult HelicopterSubclass::FUN_100040a0(Vector4Impl& p_v, float p_f)
 {
-	return SUCCESS;
+	MxU32 state = m_unk0x30;
+	if (state == 1) {
+		p_v.EqualsImpl(m_unk0x0.GetVector().elements);
+		p_v[3] = acos(p_v[3]) * (1 - p_f) * 2.0;
+		p_v.NormalizeQuaternion();
+		return SUCCESS;
```
Comment (foxtacles) [inline LEGO1/helicopter.cpp:332]: This should return the result of `NormalizeQuaternion`

File: LEGO1/helicopter.cpp [hunk 2]
```diff
@@ -321,8 +321,51 @@ void Helicopter::VTable0x70(float p_float)
...
+	}
+	else if (state == 2) {
+		p_v.EqualsImpl(m_unk0x18.GetVector().elements);
+		p_v[3] = acos(p_v[3]) * p_f * 2.0;
+		p_v.NormalizeQuaternion();
+		return SUCCESS;
```
Comment (foxtacles) [inline LEGO1/helicopter.cpp:338]: This should return the result of `NormalizeQuaternion`

File: LEGO1/helicopter.cpp [hunk 3]
```diff
@@ -321,8 +321,51 @@ void Helicopter::VTable0x70(float p_float)
...
+			}
+			return SUCCESS;
+		}
+		p_v[0] = -m_unk0x0[1];
+		p_v[1] = m_unk0x0[1];
+		p_v[2] = -m_unk0x0[3];
+		p_v[3] = m_unk0x0[2];
+		d1 = sin((1 - p_f) * 1.570796326794895);
+		d2 = sin(p_f * 1.570796326794895);
+		for (int i = 0; i < 3; i++) {
```
Comment (foxtacles) [inline LEGO1/helicopter.cpp:364]: `MxS32` instead of `int`

File: LEGO1/helicopter.cpp [hunk 4]
```diff
@@ -321,8 +321,51 @@ void Helicopter::VTable0x70(float p_float)
...
+				double d = acos(d1);
+				sin(d);
+				d1 = sin((1 - p_f) * d) / sin(d);
+				d2 = sin(p_f * d) / sin(d);
+			}
+			else {
+				d1 = 1 - p_f;
+				d2 = p_f;
+			}
+			for (int i = 0; i < 4; i++) {
```
Comment (foxtacles) [inline LEGO1/helicopter.cpp:353]: `MxS32` instead of `int`

**Comment by foxtacles** [comment]
Gave matching a shot but couldn't get anything that avoids the consecutive jump tests at the beginning of the function - not sure what it takes. We can merge once the minor issues are fixed though.

File: LEGO1/helicopter.cpp [hunk 5]
```diff
@@ -321,8 +321,51 @@ void Helicopter::VTable0x70(float p_float)
...
+				p_v[i] = m_unk0x18[i] * d2 + m_unk0x0[i] * d1;
+			}
+			return SUCCESS;
+		}
+		p_v[0] = -m_unk0x0[1];
+		p_v[1] = m_unk0x0[1];
+		p_v[2] = -m_unk0x0[3];
+		p_v[3] = m_unk0x0[2];
+		d1 = sin((1 - p_f) * 1.570796326794895);
```
Comment (ItzSwirlz) [inline LEGO1/helicopter.cpp:362]: Maybe see if there is a constant of Pi in MSVC400 so then its 'PI / 2'

**Pull request merged**